### ISSUE TITLE
[ARCHETYPE-70] Add project description as a mojo parameter

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#490]
**[Michael Heuer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=heuermh)** opened **[ARCHETYPE-70](https://issues.apache.org/jira/browse/ARCHETYPE-70?redirect=false)** and commented

For my archetype bundle, I require the project description as a parameter.

I use it in the pom.xml

\<description>${description}\</description>

in a license HEADER.txt

/*

${artifactId}  ${description}
Copyright ...

in a package.html

\<html>
\<body>
\<p>${description}\</p>
\</body>
\</html>

and so on.

The attached patch to MavenArchetypeMojo.java provides a project description parameter in addition to project groupId, artifactId, and version:

$ mvn -X archetype:create
-DgroupId=foo
-DartifactId=bar
-Dversion=1.0-SNAPSHOT
-Ddescription="bar description." ...

---

**Attachments:**
- [patch.txt](https://issues.apache.org/jira/secure/attachment/12713733/patch.txt) (_1.04 kB_)

2 votes, 2 watchers

**Comment by jira-importer** [comment]
**[Jeff Black](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wildweasel)** commented

I would find it more flexible to be able to accept any arbitrary command line parameter and automatically include that into the velocity map.  So the archetype plugin would have its required and optional parameters, but you could also include other -D options on the command line with the archetype:create command, such as -Ddescription=\<desc> and -DprojectName=\<projectName>.

In short, I want to be able to create a fully templated prototype pom.xml with no modifications to the archetype plugin.

This would keep the archetype plugin cleaner without having to fork it locally (or patch and release a new official version) for each develope that adds an additional parameter requirement.

If this sounds reasonable, I can take a stab at contributing this solution.  Any showstoppers that I would need to be aware of.

Jeff

**Comment by jira-importer** [comment]
**[Raphael Luta](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=raphael)** commented

The future archetypeNG version will solve this issue by letting the archetype developper define a list of properties (with default values)

**Comment by jira-importer** [comment]
**[Michael Heuer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=heuermh)** commented

What about the fix for the current version?

The patch provided is very simple and causes no problems if a project description parameter is not provided.

If it would help get this fix committed, I might also be able to contribute a simple unit test, patches to documentation, etc.

**Comment by jira-importer** [comment]
**[Ernesto Barreras](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ernestob)** commented

Today I was trying to create my first archetype and was very surprised when I realized that using arbitrary parameters didn't work. I would say that this is almost a must for any not very simple archetype.

Ernesto

**Comment by jira-importer** [comment]
**[Raphael Luta](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=raphael)** commented

These issues are addressed since the 2.0-alpha-n version of the plugin

**Comment by jira-importer** [comment]
**[Michael Heuer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=heuermh)** commented

I am re-opening this issue because I still cannot see how to accomplish this, with version 2.0-alpha-3.  How about some documentation about how this was addressed?

Specifically,

1) How do you define a parameter in an archetype project that must be made accessible to Velocity templates in the archetype?

2) How do you specify a value for the new parameter when running mvn archetype:create/generate?
